inc/mailcollector.class.php

```diff
@@ -1507,14 +1508,18 @@ function getTotalMails() {
    private function getRecursiveAttached(\Laminas\Mail\Storage\Part $part, $path, $maxsize, $subject, $subpart = "") {
       if ($part->isMultipart()) {
          $index = 0;
-         foreach (new RecursiveIteratorIterator($part) as $mypart) {
+         // while instead of foreach to supress header error in laminas-mail
+         $iterator = new RecursiveIteratorIterator($part);
+         while (@$iterator->valid()) {
```

Comment:
Is that strictly required? I do not like `@` usage

Reply:
If there is an error triggered by laminas-mail on some emails, we should, at least, open an issue on their side.

I do not like suppressing errors, especially when the called method can have many unpredictible reasons to trigger an error. Indeed, you may want to suppress a specific error that may not affect us, but you will also mute all other errors related to any email parsing.

Reply:
The main problem in laminas-mails is `str_split(): Argument #2 ($length) must be greater than 0`.
I openned a PR to resolve it: https://github.com/laminas/laminas-mail/pull/128

Reply:
OK, so we'll have to wait for new laminas-mail release - or at least for that PR to be approved